src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java

```diff
@@ -451,20 +455,11 @@ public DataAccessRequest insertSubmittedDataAccessRequest(User user, String refe
      * @return The updated DataAccessRequest
      */
     public DataAccessRequest updateByReferenceId(User user, DataAccessRequest dar) {
-        Date now = new Date();
-        dataAccessRequestDAO.updateDataByReferenceId(dar.getReferenceId(),
-            user.getUserId(),
-            now,
-            dar.getSubmissionDate(),
-            now,
-            dar.getData());
-        if (Objects.nonNull(dar.getCollectionId())) {
-            darCollectionDAO.updateDarCollection(dar.getCollectionId(), user.getUserId(), now);
-        }
-        // Update the dar_dataset collection
-        syncDataAccessRequestDatasets(dar.getDatasetIds(), dar.getReferenceId());
-
-        return findByReferenceId(dar.getReferenceId());
+      try {
+        return dataAccessRequestServiceDAO.updateByReferenceId(user, dar);
+      } catch(SQLException e) {
+        throw new IllegalArgumentException("Update failed due to invalid parameters");
+      }
```

Comment:
If we hit any sql exception, we should definitely log that here as an error.
You should be able to re-throw the sql exception - those should be handled nicely in the resource layer. We generally show those to clients as an internal server error.

Reply:
Is there a clean way of re-throwing that doesn't involve having to add `throws SQLException` to functions that call `updateByReferenceId`?

Reply:
If you look at other usages in this pattern, we generally catch the sql error in the service method and do the logging there, so we could do that instead of logging it here. Then you could decide at the service layer what makes the most sense. However, no, if this method throws sql exceptions and passes them up the stack, then the stack has to deal with them.